### ISSUE TITLE
Readonly calcite-combobox not reading out "non-editable" or currently selected value

### ISSUE THREAD

**Issue by jordantsanz** [Esri/calcite-design-system#10633]
**monday.com sync:** #18048762083

### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Summary

When using the `readonly` property on the `calcite-combobox`, I see the readonly markup applied in the DOM. However, when using screenreaders (VoiceOver and NVDA tested) to read the calcite-comboboxes, the readout does not include the fact that the combobox is readonly, nor does it include the currently selected value. 

### Actual Behavior

When focusing on a [readonly combobox here](https://codepen.io/jcfranco/pen/xxevvbE), for example, I see the following readout with VoiceOver:

![Screenshot 2024-10-28 at 12 01 16 PM](https://github.com/user-attachments/assets/6f196922-b09a-4a13-b221-18a59c62709e).

This is missing that the combobox is readonly, and it is missing that the current value is "One".

### Expected Behavior

I would expect `clickable list box popup, Menu pop up combo box non-editable One` I believe, or something of the sort. 

### Reproduction Sample

https://codepen.io/jcfranco/pen/xxevvbE

### Reproduction Steps

1. Go to the codepen above and enable voiceover `command + F5` or other screenreading software.
2. Use tab to focus on a calcite-combobox and observe the readout from the screenreader.

### Reproduction Version

2.13.0

### Relevant Info

MacOS, Chrome, Safari, VoiceOver
Windows, NVDA

### Priority impact

impact - p1 - need for an upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Hub

**Comment by geospatialem** [comment]
Will explore further investigation for an upcoming a11y audit to determine next steps.

**Comment by geospatialem** [comment]
> Will explore further investigation for an upcoming a11y audit to determine next steps.

This has been added for exploration and confirmation with the 2025 third party audit of Calcite. Will move this issue to late summer for follow-up from the audit results.

**Comment by geospatialem** [comment]
Verified via the [codepen](https://codepen.io/jcfranco/pen/xxevvbE) with `4.0.0-next.30` using JAWS and NVDA, where the selected values are read out when the `combobox` component contains the `read-only` attribute.
